### PR TITLE
Add logging.

### PR DESCRIPTION
**Pull request by lordclips**
Switch internal print statements to logging for more explicit note of what is happening when and where. Ignores file using Click.

### CODE REVIEW

**Comment by lordclips** [comment]
(A few) things need a bit more tweaking and fixing but the skeleton is there, at the very least.

File: src/gtnh/gui.py [hunk 1]
```diff
@@ -17,6 +18,9 @@
 from gtnh.pack_downloader import download_mod, ensure_cache_dir
 from gtnh.utils import get_latest_release, get_token, load_gtnh_manifest, sort_and_write_modpack
 
+LOGGER = logging.getLogger("gui")
```
Comment (mitchej123) [inline src/gtnh/gui.py:21]: nit: all caps should be constants, I'd prefer `log =` and use `log.info`

Comment (lordclips) [reply src/gtnh/gui.py]: Should be fixed

**Comment by mitchej123** [comment]
needs to be rebased

**Comment by Dream-Master** [comment]
@lordclips

**Comment by lordclips** [comment]
Whoops I asked in the discord, missed this notif. Doing soon

**Pull request closed without merging**